op-devstack/dsl/ecotone_fees.go

```diff
@@ -58,24 +58,27 @@ func (ef *EcotoneFees) ValidateTransaction(from *EOA, to *EOA, amount *big.Int)
 	vaultsAfter := ef.getVaultBalances(client)
 	vaultIncreases := ef.calculateVaultIncreases(vaultsBefore, vaultsAfter)
 
-	l1Fee := big.NewInt(0)
-	if receipt.L1Fee != nil {
-		l1Fee = receipt.L1Fee
-	}
+	// In Ecotone, L1 fee includes both base fee and blob base fee components
+	l1Fee := vaultIncreases.L1FeeVault // Use actual vault increase as the source of truth
 
-	block, err := client.InfoByHash(ef.ctx, receipt.BlockHash)
-	ef.require.NoError(err)
+	// Calculate L2 fees
+	baseFee := vaultIncreases.BaseFeeVault       // Use actual vault increase as the source of truth
+	priorityFee := vaultIncreases.SequencerVault // Use actual vault increase as the source of truth
+	l2Fee := new(big.Int).Add(baseFee, priorityFee)
 
-	baseFee := new(big.Int).Mul(block.BaseFee(), big.NewInt(int64(receipt.GasUsed)))
-	l2Fee := new(big.Int).Mul(receipt.EffectiveGasPrice, big.NewInt(int64(receipt.GasUsed)))
-	priorityFee := new(big.Int).Sub(l2Fee, baseFee)
-	totalFee := new(big.Int).Add(l1Fee, l2Fee)
+	// Total fee is the sum of all vault increases
+	totalFee := new(big.Int).Add(vaultIncreases.BaseFeeVault, vaultIncreases.L1FeeVault)
+	totalFee.Add(totalFee, vaultIncreases.SequencerVault)
+	totalFee.Add(totalFee, vaultIncreases.OperatorVault)
```

Comment:
My understanding is that the operator fee was added at isthmus, after ecotone. So do not need the `OperatorVault` check at all. 
Ref) https://specs.optimism.io/protocol/isthmus/predeploys.html#operatorfeevault

We should check the increment `vaultIncreases.OperatorVault` must be zero at ecotone.